pallets/pallet-bank/src/lib.rs

```diff
@@ -434,24 +434,40 @@ pub mod module {
 			Ok(())
 		}
 
+		/// Migrate the old treasury account to a new one.
+		///
+		/// Requires Root.
 		#[pallet::call_index(8)]
 		#[pallet::weight(T::WeightInfo::rotate_treasury())]
 		pub fn rotate_treasury(origin: OriginFor<T>, new_treasury: T::AccountId) -> DispatchResult {
 			ensure_root(origin)?;
 			ensure!(!Accounts::<T>::contains_key(&new_treasury), Error::<T>::AccountIdAlreadyTaken);
-			let old_treasury = Self::treasury()?;
-			// Update lock expiry of old treasury to the new treasury account.
-			AccountWithUnlockedFund::<T>::iter().for_each(|(block_number, mut accounts)| {
-				accounts.iter_mut().for_each(|(user_id, _)| {
-					if *user_id == old_treasury {
-						*user_id = new_treasury.clone();
-					}
-				});
-				// Update the storage map with the modified accounts
-				AccountWithUnlockedFund::<T>::insert(block_number, accounts);
-			});
+			ensure!(
+				T::RoleManager::role(&new_treasury).is_none(),
+				Error::<T>::AccountIdAlreadyTaken
+			);
+
+			let old_treasury = match Self::treasury() {
```

Comment:
I think it would be slightly better to use 
`let old_treasury = Self::treasury().ok()`
 to change from Result -> Option first
then
`if let Some(treasury) = old_treasury {
...
}`